### ISSUE TITLE
Techno Bars

### ISSUE THREAD

**Issue by apibrac** [yolanlepibrac/website-portefolio#1]
Je te met un peu plus que ton pb au hover

**Comment by apibrac** [comment]
Ton fichier est vraiment trop grand !
C'est le genre de truc super dûr à maintenir : genre il suffit que tu décide de tout wrapper dans une div ou changer la moindre conditions de rendu à haut niveau et tu dois t'assurer que tout fonctionne correctement
En plus les éléments qui concerne la même partie sont séparés en plusieurs entre le state à un endroit, les handlers correspondant ailleurs, la fonction de rendu encore ailleurs et le black dans le jsx

https://github.com/yolanlepibrac/website-portefolio/blob/064149c0b435ad07e3fe8f54d5fb6fd6f80f9c10/src/App.js#L402

ça par exemple ça ferrait un parfait composant, que tu met dans un autre fichier avec son state, ses handlers et tout

**Comment by apibrac** [comment]
https://github.com/yolanlepibrac/website-portefolio/blob/064149c0b435ad07e3fe8f54d5fb6fd6f80f9c10/src/App.js#L69

Ici ça se voit que ton composant est dur à utiliser avec les top1 et top2 etc
ça serait plus classe d'avoir juste une valeur top (d'ailleurs dans ce genre de composant c'est mieux de pas géré la position si tu gère déjà le rendu / la taille, il vaut mieux laisser le parent faire)
et au lieu de top2 faire `top - 5` ou + ou peut importe (ou alors pas 5 mais utiliser un autre param directement)

**Comment by apibrac** [comment]
(juste pour info ta fonctionnalité précédante peut être faite avec une propriété css `scale` sur l'élement, avec la transition sur le hover qui va bien)

**Comment by apibrac** [comment]
Du coup ton problème pour les bars c'est que pour une raison ou une autre la couleur (propriété background color) ne suit plus, elle est settée à un truc random transparent et du coup on voit plus les bars

Dommage qu'on ait pas fait la track du bug ensemble mais il fallait inspecter et voir la barre changer de propriété au hover, et voir la différence entre celles qui sont encore visibles et celles qui déconnent

Bref du coup normalement t'as pas besoin de changer la couleur de bars, tu devrait pouvoir les laisser blanches et plutôt bosser sur l'opacité ou sur leur largeur pour qu'on les voit plus

**Comment by yolanlepibrac** [comment]
Jvais pas pouvoir regarder tout ce soir, mais j'ai checké vite fait ce
point.  Effectivement, j'ai vu le debogger, elles deviennent de couleur
rgba(45,45,45,0). C'est vrai que mon pose agit sur la couleur et la largeur
:
idle: {
width : 0,
backgroundColor:'rgba(0,0,0,0)',
transition: ({i}) => ({
duration: 300,
ease: 'backInOut',
delay: i*0,
})
},
hovered: {
width:({width})=>(width),
backgroundColor:'rgba(255,255,255,1)',
transition: ({i}) => ({
duration: 100,
ease: 'easeOut',
delay: i*100,
})

Du coup, je peux tester de passer la couleur de 'rgba(255,255,255,0)' a '
rgba(255,255,255,1)', pour laisser la couleur blanche. Je sais pas si c'est
possible de jouer sur l'opacité d'un div autrement
que par la couleur (je crois que j'avais déjà essayer, j'avais pas trouvé).

**Comment by apibrac** [comment]
si si t'as la propriété css "opacity"

mais est ce que t'as vraiment besoin de faire varier l'opacity ?
si la width est à zero, l'élément sera déjà plus visible non ?
ah tu veux que l'opacity change graduellement ?

**Comment by yolanlepibrac** [comment]
Yes cetait ça l'idée, pour avoir un effet d'apparition en plus. Mais
peut-être pas nécessaire. Yes jai vu ça, opacity. Je teste tout ça ce soir,
et je te dis !
